main/src/library/Console.flix

```diff
@@ -91,6 +91,43 @@ namespace Console {
                 esc + "[38;2;1m" + ToString.toString(s) + esc + "[0m"
         }
 
+    ///
+    /// Returns `s` formatted with color `hexCode`
+    /// or if there's an error, returns the original string.
+    /// Expected format of `hexCode`: "#" + "--" + "--" + "--"
+    /// where each -- is a hex number between 00 and FF.
+    ///
+    pub def hex(hexCode: String, s: a): String with ToString[a] =
+        let text = ToString.toString(s);
+        match String.stripPrefix(substr = "#", hexCode) {
+            case Some(hex) => {
+                if (String.length(hex) == 6) {
+                    let red = String.sliceLeft(2, hex);
+                    let green = String.slice(2, 4, hex);
+                    let blue = String.slice(4, 6, hex);
+
+                    match (stringToHex(red), stringToHex(green), stringToHex(blue)) {
+                        case (Some(r), Some(g), Some(b)) => rgb((r, g, b), text, isFg = true)
+                        case _                           => text
+                    }
+                } else text
+            }
+            case _ => text
```

Comment:
I could not get the `let*` to work, so this ended up a bit ugly...

Reply:
This isn't _too_ bad :)

But for informational/educational purposes:

The basic idea is that everything that returns an Option, we assign with a `let*`, which collapses all the `None`s together.

Here's an untested example
```scala
...
let text = ToString.toString(s);
let hexOpt: Option[String] = {
  use Option.flatMap;
  let* hex = String.stripPrefix(substr = "#", hexCode);
  if (String.length(hex) == 6) {
    let* r = String.sliceLeft(2, hex) |> stringToHex;
    let* g = String.slice(2, 4, hex) |> stringToHex;
    let* b = String.slice(4, 6) |> stringToHex;
    Some(rgb((r, g, b), text, isFg = true))
  } else None
};
hexOpt |> Option.getWithDefault(text)

Reply:
Yeah, that it much nicer! I think I will have to remember to use `|>` a bit more as well.